src/main/java/wolox/training/controllers/UsersController.java

```diff
@@ -0,0 +1,81 @@
+package wolox.training.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.IdMismatchException;
+import wolox.training.exceptions.DataNotFoundException;
+import wolox.training.exceptions.NotificationCode;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("/api/users")
+public class UsersController {
+
+	@Autowired
+	private UserRepository usersRepository;
+
+	/**
+	 * This method returns all users
+	 * @return List<Iterable> the User
+	 */
+	@GetMapping
+	public Iterable findAll() {
+		return usersRepository.findAll();
+	}
+
+	/**
+	 * This method returns all users by username
+	 * @param username: author of the user (String)
+	 * @return  {@link User}
+	 */
+	@GetMapping("/name/{name}")
+	public User findByName(@PathVariable String username) {
```

Comment:
I'd prefer to call this method findByUsername and the endpoint also to be /username/{username}, because a user can have both name and username, so it may confuse the client who is using it.